### PR TITLE
GEODE-10126: Replace Geode for Redis properties with system properties

### PR DESCRIPTION
**Pull request by jdeppe-pivotal**
- All previous Geode for Redis properties that were defined in
  geode-core are replaced by java system properties.
  - gemfire.geode-for-redis-port
  - gemfire.geode-for-redis-bind-address
  - gemfire.geode-for-redis-username
  - gemfire.geode-for-redis-redundant-copies
  - gemfire.geode-for-redis-enabled
- Redis is enabled (using defaults) if the `enabled` property is set to
  true OR if any of the other properties are explicitly set as java
  system properties.
- Introduce `RedisConfiguration` interface to access properties
- Concrete implementation of `SystemPropertyBasedRedisConfiguration`
  that uses system properties to define Radish properties.
- Integration Tests use `TestRedisConfiguration` to avoid having to
  set system properties.
- Services started by Geode (`CacheService`s) that throw exceptions
  during initialization will cause the Cache to fail to startup.
  Previously any such errors were only logged.

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### CODE REVIEW

**Comment by upthewaterspout** [comment]
It looks like this changes the default bind address to 127.0.0.1. It used to default to listening on an actual network interface. Was that intentional?

**Comment by jdeppe-pivotal** [comment]
> It looks like this changes the default bind address to 127.0.0.1. It used to default to listening on an actual network interface. Was that intentional?

I thought it would be more secure but I realized that Geode servers do that in any case, so I left it as it was. However, as per some other conversations the bind address is now derived from Geode's `server-bind-address`. So the order of precedence is:
- If the java system property `gemfire.geode-for-redis-bind-address` is set, then use that as the bind address.
- If the Geode property `server-bind-address` is set, then use that.
- Use the default which just binds to all addresses.

File: geode-for-redis/src/acceptanceTest/java/org/apache/geode/redis/internal/commands/executor/GeodeRedisServerStartupUsingGfshAcceptanceTest.java
Comment (DonalEvans) [inline geode-for-redis/src/acceptanceTest/java/org/apache/geode/redis/internal/commands/executor/GeodeRedisServerStartupUsingGfshAcceptanceTest.java:154]: This test name seems incomplete.

File: geode-for-redis/src/commonTest/java/org/apache/geode/redis/TestRedisConfiguration.java
Comment (DonalEvans) [inline geode-for-redis/src/commonTest/java/org/apache/geode/redis/TestRedisConfiguration.java:90]: Should this return `redundantCopies`?

File: geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/connection/AuthIntegrationTest.java
Comment (DonalEvans) [inline geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/connection/AuthIntegrationTest.java:128]: I think the `build()` at the end of this is not necessary here.

File: geode-for-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisServer.java
Comment (DonalEvans) [inline geode-for-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisServer.java:71]: This comment should not reference the address and port, but rather the `RedisConfiguration`.

**Review by DonalEvans — APPROVED** [review]
Just a few small things to clean up, but nothing that should block the PR

**Review by kamilla1201 — APPROVED** [review]
The files I'm code owner for look good

File: geode-for-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisService.java
Comment (jdeppe-pivotal) [reply geode-for-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisService.java:93]: I think this all ties in to handling a reconnect where the cache is cycled and services are re-initialized. This happens because `GeodeRedisService` is a `ResourceEventsListener`. The restart is triggered by `GemfireCacheImpl.notifyResourceListeners`. In that method various exceptions are caught and logged, but only `GemFireSecurityException` and `ManagementException` cause an actual failure. Hence why we wrap in `ManagementException`. It's not pretty I know... We probably need to have `ResourceEventsListener.handleEvent` throw a specific exception if we want the problem/exception to bubble up.

File: geode-for-redis/src/acceptanceTest/java/org/apache/geode/redis/internal/commands/executor/GeodeRedisServerStartupUsingGfshAcceptanceTest.java
Comment (jdeppe-pivotal) [reply geode-for-redis/src/acceptanceTest/java/org/apache/geode/redis/internal/commands/executor/GeodeRedisServerStartupUsingGfshAcceptanceTest.java]: Updated to `gfshStartsRedisServer_withAnyLocalBindAddress`.

File: geode-for-redis/src/commonTest/java/org/apache/geode/redis/TestRedisConfiguration.java
Comment (jdeppe-pivotal) [reply geode-for-redis/src/commonTest/java/org/apache/geode/redis/TestRedisConfiguration.java]: Fixed.

File: geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/connection/AuthIntegrationTest.java
Comment (jdeppe-pivotal) [reply geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/connection/AuthIntegrationTest.java]: Yes, you're correct.

File: geode-for-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisServer.java
Comment (jdeppe-pivotal) [reply geode-for-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisServer.java]: Updated.

**Pull request merged**
